### ISSUE TITLE
Adding a CHANGELOG

### ISSUE THREAD

**Issue by bartocc** [ethereum-mining/ethminer#631]
A CHANGELOG would be a great addition to the project.

As of today, how can I know the differences between 0.12 and 0.13 ?

Thanks

**Comment by chfast** [comment]
Review merged PRs.

**Comment by jramos** [comment]
Use `git changelog` from [`git-extras`](https://github.com/tj/git-extras/blob/master/Installation.md).

```bash
$ git changelog --merges-only -s v0.12.0 -f v0.13.0

v0.13.0rc9 / 2018-01-22
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
  * Merge pull request #599 from jean-m-cyr/switch
    CUDA: reduce job switch time to sub 10 milliseconds
  * Merge pull request #602 from ethereum-mining/circleci
    Circle CI: Upgrade CUDA docker base image
  * Merge branch 'master' into switch
  * Merge branch 'master' into switch

v0.13.0rc8 / 2018-01-21
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
  * Merge pull request #571 from nerdralph/master
    match stale solutions to correct header
  * Merge pull request #598 from jagerman/long-jobno-fix
    Fix ethereumstratum when job number is >8 chars
  * Merge pull request #594 from jean-m-cyr/master
    URGENT patch. Fix bad merge.
  * Merge pull request #591 from jean-m-cyr/cm
    Merge both major cuda miners classes into one.
  * Merge branch 'master' into cm
  * Merge pull request #588 from jean-m-cyr/i563
    I563
  * Merge pull request #582 from tadeegan/patch-1
    Fix up indentation in FAQ readme.
  * Merge pull request #578 from rushstrike/master
    make sure to initialize the correct number of CUDA devices
    in mixed mode -X param
  * Merge branch 'master' of https://github.com/nerdralph/ethminer
  * Merge pull request #1 from ethereum-mining/master
    sync with upstream
  * Merge pull request #575 from MariusVanDerWijden/opencl
    Opencl: Test whether DAG fits in memory everytime we
    recreate it
  * Merge pull request #574 from MariusVanDerWijden/master
    fixed #464; Position of -RH option

v0.13.0rc7 / 2018-01-15
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
  * Merge pull request #560 from ethereum-mining/cuda-reporting
    CUDAMiner: Pass work package to report()
  * Merge pull request #556 from ethereum-mining/catch-exceptions
    Catch exceptions from execute()

v0.13.0rc6 / 2018-01-12
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
  * Merge pull request #547 from jean-m-cyr/nvidia-switch
    Halve the nvidia farm switch time.

v0.13.0rc5 / 2018-01-11
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
  * Merge pull request #534 from jgillich/master
    fix _WIN32 ifdef
  * Merge pull request #525 from jean-m-cyr/stratum2
    Apply work timer fix to stratum v2
  * Merge pull request #523 from jean-m-cyr/issue505
    Issue505 - provide pool addresses to API
  * Merge pull request #519 from jean-m-cyr/sections
    Strip unused (dead) code and data sections
  * Merge pull request #521 from jean-m-cyr/timeout
    IO timeout not working. Failover and reconnect not
    happening!

v0.13.0rc4 / 2018-01-08
=======================
  * Merge remote-tracking branch 'origin/master' into release/0.13
    # Conflicts:
    #	libethash-cuda/CUDAMiner.cpp
  * Merge pull request #515 from EoD/restore_clover_support_for_unstable_kernel
    libethash-cl: restore Mesa/Clover support in unstable kernel
  * Merge pull request #506 from EoD/restore_old_cl_kernel
    Restore old cl kernel v4
  * Merge pull request #509 from birdayz/bugfix/clminer_init_race_condition
    Fix race condition in CLMiner::init through mutex
  * Merge pull request #501 from MariusVanDerWijden/master
    Check whether the device has enough RAM whenever we create
    the DAG
  * Merge pull request #474 from ethereum-mining/cudaminer
    Cleanup CUDA Miner
  * Merge remote-tracking branch 'origin/release/0.12'
    # Conflicts:
    #	.bumpversion.cfg
    #	CMakeLists.txt
  * Merge pull request #498 from miscellaneousbits/master
    Honor the -L single <n> cmd line parameter
  * Merge pull request #496 from jean-m-cyr/master
    No duplicate evals, no invalid shares, minimal diff
  * Merge pull request #487 from muraj/master
    Fix segfault and deadlock in CUDAMiner.
  * Merge pull request #494 from EoD/rework_cl_error_printing
    Rework OpenCL error printing
  * Merge pull request #473 from ethereum-mining/cuda-parallel-hash
    CUDA: Limit number of options for parallel hashes
  * Merge pull request #361 from MariusVanDerWijden/branch_#359
    Optimise work updates on CUDA, was Branch#359
  * Merge branch 'master' into branch_#359
  * Merge branch 'master' into branch_#359
  * Merge pull request #468 from jean-m-cyr/master
    Clean up and minor optimizations
  * Merge branch 'master' of https://github.com/ethereum-mining/ethminer
  * Merge pull request #471 from ethereum-mining/hunter
    Upgrade Hunter
  * Merge branch 'branch_#359' of https://github.com/MariusVanDerWijden/ethminer into branch_#359
  * Merge pull request #430 from smurfy/hwmon2
    Fixed free vs delete
  * Merge pull request #404 from lev-kusanagi/compute-capability-7
    Add compatibility with compute capability 7.0
  * rebase from Master
  * Merge branch 'master' into branch_#359
  * Merge pull request #416 from jean-m-cyr/master
    10 digit target always 0 in log messages!!!
  * Merge branch 'master' into branch_#359
  * Merge pull request #407 from ethereum-mining/cuda9
    AppVeyor: Install CUDA 9
  * Merge pull request #373 from djwbrown/nvidia-jetsontx1
    Added CUDA compute_53 to support tegra X1.
  * Merge pull request #319 from smurfy/hwmon
    Hardware Monitoring for NVML, ADL and AMDGPU SysFS
  * Merge pull request #400 from MariusVanDerWijden/sync
    fixed dumb error
  * Merge pull request #398 from ethereum-mining/cuda9
    Tracis CI: Upgrade CUDA to 9.0
  * Merge pull request #353 from MariusVanDerWijden/sync
    added support for __shfl_sync since __shfl is deprecated
    with CUDA 9.0
  * Merge pull request #396 from ethereum-mining/travis-macos
    Travis CI: Upgrade Xcode
  * Merge pull request #367 from mbajkowski/patch-1
    Update to fix stratum failover argument handling
  * Merge pull request #294 from goobur/new-kernel
    New OpenCL kernel
```

**Comment by DeadManWalkingTO** [comment]
I think this issue can be closed.
